src/main/java/mil/dds/anet/database/PositionDao.java

```diff
@@ -576,6 +593,20 @@ public int mergePositions(Position winner, Position loser) {
     updateM2mForMerge("authorizationGroupPositions", "authorizationGroupUuid", "positionUuid",
         winnerUuid, loserUuid);
 
+    // Update organizationAdministrativePositions
+    deleteForMerge("organizationAdministrativePositions", "positionUuid", loserUuid);
+    deleteForMerge("organizationAdministrativePositions", "positionUuid", winnerUuid);
+
+    List<Organization> winnerOrganizations = winner.getOrganizationsAdministrated();
+    final Set<String> winnerOrgUuids = winnerOrganizations == null ? Collections.emptySet()
+        : winnerOrganizations.stream().map(ap -> ap.getUuid()).collect(Collectors.toSet());
+    for (String uuid : winnerOrgUuids) {
+      getDbHandle().createUpdate(
+          "INSERT INTO \"organizationAdministrativePositions\"(\"organizationUuid\", \"positionUuid\")"
+              + "VALUES(:organizationUuid, :positionUuid)")
+          .bind("organizationUuid", uuid).bind("positionUuid", winnerUuid).execute();
+    }
```

Comment:
Shouldn't this be simply something like:
```
    updateM2mForMerge("organizationAdministrativePositions", "organizationUuid", "positionUuid", winnerUuid, loserUuid);
```
?

Reply:
Ah, OK; in that case there's a utility method you could use here:
```java
@@ -595,17 +597,9 @@ public class PositionDao extends AnetSubscribableObjectDao<Position, PositionSea
 
     // Update organizationAdministrativePositions
     deleteForMerge("organizationAdministrativePositions", "positionUuid", loserUuid);
-    deleteForMerge("organizationAdministrativePositions", "positionUuid", winnerUuid);
-
-    List<Organization> winnerOrganizations = winner.getOrganizationsAdministrated();
-    final Set<String> winnerOrgUuids = winnerOrganizations == null ? Collections.emptySet()
-        : winnerOrganizations.stream().map(ap -> ap.getUuid()).collect(Collectors.toSet());
-    for (String uuid : winnerOrgUuids) {
-      getDbHandle().createUpdate(
-          "INSERT INTO \"organizationAdministrativePositions\"(\"organizationUuid\", \"positionUuid\")"
-              + "VALUES(:organizationUuid, :positionUuid)")
-          .bind("organizationUuid", uuid).bind("positionUuid", winnerUuid).execute();
-    }
+    Utils.addRemoveElementsByUuid(existingPos.loadOrganizationsAdministrated(context).join(),
+        winner.getOrganizationsAdministrated(), newOrg -> addOrganizationToPosition(winner, newOrg),
+        oldOrgUuid -> removeOrganizationFromPosition(oldOrgUuid, winner));
 
     // Update customSensitiveInformation for winner
     DaoUtils.saveCustomSensitiveInformation(null, PositionDao.TABLE_NAME, winnerUuid,
```
(with `context` = `AnetObjectEngine.getInstance().getContext()` declared higher up as it's now used multiple times)

Reply:
The organizations picked in the merge will be in the `winner.getOrganizationsAdministrated()` so these will get added by `Utils.addRemoveElementsByUuid()`.